src/parse.rs

```diff
@@ -71,6 +71,54 @@ where
     }
 }
 
+pub(crate) enum MapOrNot2<'a, T, U> {
+    Map1(T),
+    Map2(U),
+    Not(Response<'a>),
+    #[allow(dead_code)]
+    Ignore,
+}
+
+/// Parse many `T` or `U` Responses with `F` and extend `into1` or `into2` with them.
+/// Responses other than `T` or `U` go into the `unsolicited` channel.
+pub(crate) fn parse_many_into2<'input, T, U, F>(
```

Comment:
Now that we have this, we can make `parse_many_into` just call this one with `()` (or even `!`) as `U`.